h2o-core/src/main/java/hex/Model.java

```diff
@@ -268,6 +267,9 @@ public void setTrain(Key<Frame> train) {
       final boolean _needResponse;
       boolean needsResponse() { return _needResponse; }
     }
+    
+    public Key<ModelPreprocessor>[] _preprocessors;
```

Comment:
Do we want the preprocessors to have an explicit order ? Or do we want to rely on serialization/deserialization order and iterations ?

Reply:
@Pscheidl if we apply multiple TE models, for example te1 apply to some col1 with params1, te2 applies to col2 with params2, and te3 applies to grouped columns te3+te4... then these TE applications are not fully commutative as they generate new columns, and we must guarantee that the order of those columns is the same during training and in the mojo.
The easiest way to guarantee this order currently is to apply the various TE models in the same order in the Mojo.

So yeah, explicit order is important.

Reply:
Note that the serialization order that you mention is the same as the order of preprocessors applied during training.